File: mssql/csharp/read/mssql_read.cs
// ---------------------------------------------------------------
/*
	read/mssql_read.cs

				Nov/18/2015
*/
// ---------------------------------------------------------------
using System;
using System.Text;
using System.Data.SqlClient;

// ---------------------------------------------------------------
class mssql_read
{
// ---------------------------------------------------------------
static void Main (string [] args)
{
	SqlConnection cnn;
	SqlCommand cmd;

	Console.WriteLine ("*** 開始 ***");

	string server_mssql = @"server=host_mssql\EG;";

	string str_connect = server_mssql
			+ "uid=sa;"
			+ "pwd=Tiger123;"
			+ "database=city";

	cnn = new SqlConnection (str_connect);
	cnn.Open ();

	string str_sql = "Select id,name,population,date_mod From cities";

	cmd = new SqlCommand (str_sql, cnn);

	command_exec (ref cmd);
	cnn.Close ();

	Console.WriteLine ("*** 終了 ***");
}

// ---------------------------------------------------------------
static void command_exec (ref SqlCommand cmd)
{
	SqlDataReader dr_sql;
	dr_sql = cmd.ExecuteReader();

	while (dr_sql.Read ())
	{
	StringBuilder str_ss = new StringBuilder ();

	for (int it = 0; it < dr_sql.FieldCount; it++)
		{
		str_ss.Append (dr_sql.GetValue(it).ToString ());
		str_ss.Append ("\t");
		}

	Console.WriteLine (str_ss);
	}

	dr_sql.Close();
}

// ---------------------------------------------------------------
}

// ---------------------------------------------------------------

File: mssql/vb/read/mssql_read.vb
' ---------------------------------------------------------------
'
'	read/mssql_read.vb
'
'					Aug/24/2015
'
' ---------------------------------------------------------------
imports System
imports System.Text
imports System.Data.SqlClient

Module mssql_read
' ---------------------------------------------------------------
Sub Main ()

	Console.WriteLine("*** 開始 ***")

	Dim server as String ="host_mssql"
	Dim str_db as String ="city"

	Dim str_connect as String = "server=host_mssql;" _
			+ "uid=sa;" _
			+ "pwd=scott_tiger;" _
			+ "database=city"


	Dim conn as SqlConnection = new SqlConnection (str_connect)

	conn.Open()

	Dim str_sql as String = "Select id,name,population,date_mod From cities"

	Dim command as SqlCommand = new SqlCommand (str_sql, conn)

	Dim dr as SqlDataReader = command.ExecuteReader()

	Dim delim as String = vbTab

	While dr.Read()
		For it As Integer = 0 To (dr.FieldCount - 1) 
			Console.Write (dr.GetValue(it).ToString ())
			Console.Write (delim)
		Next
		Console.WriteLine ()

	End While

	conn.Close ()

	Console.WriteLine ("*** 終了 ***")

End Sub
' ---------------------------------------------------------------
End Module
' ---------------------------------------------------------------
